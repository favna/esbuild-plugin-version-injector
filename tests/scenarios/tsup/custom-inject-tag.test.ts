import tsup, { type Options } from 'tsup';
import type { PluginOptions } from '../../../src';
import { assertFileContent, buildAbsolutePath, createTsupConfig } from '../../common/util';

function createConfig(pluginOptions?: PluginOptions): Options {
  return createTsupConfig(
    {
      format: ['esm'],
      entry: [buildAbsolutePath('../fixtures/build-in/custom-inject-tag.mts')]
    },
    pluginOptions
  );
}

describe.only('Custom Version Inject Tag', () => {
  test('GIVEN custom tag THEN injects version', async () => {
    const config = createConfig({
      injectTag: '[VersionInject][/VersionInject]'
    });

    await tsup.build(config);

    await assertFileContent('../fixtures/build-out/tsup/custom-inject-tag.mjs');
  });
});
