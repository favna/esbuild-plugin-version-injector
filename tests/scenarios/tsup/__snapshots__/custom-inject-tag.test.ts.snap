// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Custom Version Inject Tag > GIVEN custom tag THEN injects version 1`] = `
"// tests/fixtures/build-in/custom-inject-tag.mts
var version = "1.2.1";

export { version };
"
`;
